cmd/kubeadm/app/util/caddyfile/json.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```

Comment:
I do think, that, at least, the copyright should be the original one, since we did not do any changes to any copied code.

Reply:
+1 for using the old copyright.

i think we should just copy json.go (not even json_test.go)
https://github.com/mholt/caddy/tree/master/caddyfile

the problem with this package is that it parse.go pulls https://github.com/mholt/caddy/tree/master/caddyfile
which don't need.

Reply:
@rosti 
Thanks , I agree with you. In order to avoid legal risk, let's leave things as they are.
@neolit123 
This doesn't seem to affect kubeadm to be in staging.

Reply:
@SataQiu the problem i see here is that we are pulling a big package for 100 lines.
see parse.go / "github.com/mholt/caddy/telemetry"

@chrisohaver @rajansandeep 
WDYT we should do here:
- continue importing caddyfile
- copy caddyfile json,*.go under kubeadm.
- rewrite the json functions

let's get the take of coredns maintainers again, before closing.

/reopen

Reply:
Ah... the redundancy vs dependency debate ...

> WDYT we should do here

dunno - probably safe code wise to copy the functions locally since these functions have not changed in a long time.  Legally is another matter - i guess you'd want to put the copied  functions (two i think) in a separate file,  duplicate the original copyright, and put a comment in there about where/when they came from.

Reply:
i also this this is the correct action here. thanks for the response.
@kubernetes/sig-cluster-lifecycle-pr-reviews comments from others?

Reply:
Another question is, how can we pass the `boilerplate` check ?
When I add the following to `cmd/kubeadm/app/util/caddyfile/json.go`:
```
// Copyright 2015 Light Code Labs, LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```

```
$ ./hack/verify-boilerplate.sh
Boilerplate header is wrong for: /home/work/src/k8s.io/kubernetes/cmd/kubeadm/app/util/caddyfile/json.go
```

Reply:
> Another question is, how can we pass the boilerplate check ?

Ugh.  Good point.

Reply:
Might be good to add a whitelist for files to skip in the boilerplate check.